rotkehlchen/db/schema.py

```diff
@@ -110,6 +110,8 @@
 INSERT OR IGNORE INTO location(location, seq) VALUES ('l', 44);
 /* Bybit */
 INSERT OR IGNORE INTO location(location, seq) VALUES ('m', 45);
+/* Scroll */
+INSERT OR IGNORE INTO location(location, seq) VALUES ('n', 46);
```

Comment:
This addition also needs to go to the DB upgrade as for already existing DBs it will be missing. Check older upgrades. For example in v40_v41 we got this.
```
def _add_new_supported_locations(write_cursor: 'DBCursor') -> None:
    log.debug('Enter _add_new_supported_locations')
    write_cursor.execute(
        'INSERT OR IGNORE INTO location(location, seq) VALUES (?, ?)',
        ('m', 45),
    )
    log.debug('Exit _add_new_supported_locations')
    ```

You need to put this in the current upgrade and test for it. Chech the `test_db_upgrades.py` to see how. You just need to add to the latest upgrade test. This upgrade will hit users with release 1.33.0.

Reply:
There is no such way. The DB upgrades can't be made automatically as they are all custom code each time and the most sensitive part of upgrading to a new version.

It's easy to do if you have any questions ask in discord.

Reply:
Not saying that it isn't easy, once I have done it once, or someone has shown it to me. But I imagine, if we want to add 10-20 new chains this year, and perhaps enroll 10 new contributors, this will be a pain in the a** for them :)

Reply:
no way around it I am afraid. Open to concrete architectural suggestions to improve modularity.

Reply:
addressed in https://github.com/rotki/rotki/pull/7584/commits/a72c9e4587bf541c44eb6f57f50aed3250a07ae2